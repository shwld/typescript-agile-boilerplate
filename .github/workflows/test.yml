name: test

on:
  push:
    branches: ["main"]
  pull_request:
    types: [opened, synchronize]

jobs:
  build:
    name: Test
    timeout-minutes: 15
    runs-on: ${{ matrix.os }}
    # To use Remote Caching, uncomment the next lines and follow the steps below.
    # env:
    #  TURBO_TOKEN: ${{ secrets.TURBO_TOKEN }}
    #  TURBO_TEAM: ${{ secrets.TURBO_TEAM }}
    strategy:
      matrix:
        # os: [ubuntu-latest, macos-latest]
        os: [ubuntu-latest]
    services:
      # Label used to access the service container
      postgres:
        # Docker Hub image
        image: postgres
        # Provide the password for postgres
        env:
          POSTGRES_USER: postgres
          POSTGRES_PASSWORD: password
          POSTGRES_DB: app_test
        # Set health checks to wait until postgres has started
        options: >-
          --health-cmd pg_isready
          --health-interval 10s
          --health-timeout 5s
          --health-retries 5

    steps:
      - name: Check out code
        uses: actions/checkout@v2
        with:
          fetch-depth: 2

      - name: Setup Node.js environment
        uses: actions/setup-node@v2
        with:
          node-version: 16
          cache: 'yarn'

      - name: Setup Test
        run: |
          echo 'DATABASE_URL="postgresql://postgres:password@postgres:5432/app_test?schema=public"' > ./infrastructures/db/.env
          echo 'DATABASE_URL="postgresql://postgres:password@postgres:5432/app_test?schema=public"' > ./use-cases/graphql-resolvers/.env
          yarn db:migrate

      - name: Install dependencies
        run: yarn

      - name: Test
        run: yarn test
