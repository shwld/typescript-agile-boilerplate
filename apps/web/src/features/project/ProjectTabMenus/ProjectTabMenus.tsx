import {
  Box,
  BoxProps,
  Flex,
  HStack,
  Text,
  TextProps,
  useColorModeValue,
} from '@chakra-ui/react';
import { FC, ReactNode } from 'react';
import ActiveLink from '~/components/ActiveLink/ActiveLink';

const TabMenu: FC<{ isActive?: boolean; children: ReactNode } & TextProps> = ({
  isActive,
  children,
  ...props
}) => (
  <Text
    borderBottom={isActive ? '2px' : '0'}
    borderBottomColor="blue"
    px="1"
    cursor="pointer"
    {...props}
  >
    {children}
  </Text>
);

export const ProjectTabMenus: FC<{ projectId: string } & BoxProps> = ({
  projectId,
  ...props
}) => {
  return (
    <Box
      bg={useColorModeValue('blue.200', 'gray.900')}
      px={4}
      mb={2}
      {...props}
    >
      <Flex h={6} alignItems={'center'} justifyContent={'space-between'}>
        <HStack as={'nav'} spacing={4} display={{ base: 'none', md: 'flex' }}>
          <ActiveLink href={`/projects/${projectId}`}>
            <TabMenu>STORIES</TabMenu>
          </ActiveLink>
          <ActiveLink href={`/projects/${projectId}/members`}>
            <TabMenu>MEMBERS</TabMenu>
          </ActiveLink>
        </HStack>
      </Flex>
    </Box>
  );
};
