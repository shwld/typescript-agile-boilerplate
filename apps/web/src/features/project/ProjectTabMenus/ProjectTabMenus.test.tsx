import { ProjectTabMenus } from './ProjectTabMenus';
import { render } from '@testing-library/react';
import { MockedRouterProvider } from '~/test/MockedRouterProvider';

describe('ProjectTabMenus', () => {
  const renderComponent = () => {
    const renderResult = render(
      <MockedRouterProvider>
        <ProjectTabMenus projectId="test" />
      </MockedRouterProvider>
    );
    return renderResult;
  };
  test('Snapshot', () => {
    expect(renderComponent().asFragment()).toMatchSnapshot();
  });
  test('success', () => {
    const { getByText } = renderComponent();
    expect(getByText('STORIES')).toBeTruthy();
  });
});
