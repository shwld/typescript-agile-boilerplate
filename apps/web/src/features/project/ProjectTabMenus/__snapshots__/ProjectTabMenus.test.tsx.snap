// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProjectTabMenus Snapshot 1`] = `
<DocumentFragment>
  <div
    class="css-1c8n7xu"
  >
    <div
      class="css-72r3fu"
    >
      <nav
        class="chakra-stack css-t14v28"
      >
        <p
          class="chakra-text css-8qb92z"
        >
          STORIES
        </p>
        <p
          class="chakra-text css-8qb92z"
        >
          MEMBERS
        </p>
      </nav>
    </div>
  </div>
</DocumentFragment>
`;
